src/Webhook.jl

```diff
@@ -0,0 +1,151 @@
+module Webhook
+    export register_webhook!
+
+    using FromFile
+    @from "Actions.jl" using Actions
+    @from "Types.jl" using Types
+    @from "FileHelpers.jl" import FileHelpers: find_notebook_files_recursive
+    @from "Export.jl" import Export: default_index
+    import Pluto: without_pluto_file_extension
+    using HTTP
+    using SHA
+
+    # This function wraps our functions with PlutoSliderServer context. run_server & start_dir are set by the webhook options.
+    function register_webhook!(router, notebook_sessions, server_session, settings, static_dir)
+
+        """
+        Handle any events from GitHub.
+        Use with Webhook - see README for detailed HOWTO
+        This function assumes you run slider server from a GitHub repository.
+        When invoked (POST @ /github_webhook endpoint, properly authenticated)
+        the server will try to
+            - reload all changed files,
+            - stop all deleted files and
+            - start any new files
+        respecting the settings (exlusions etc.)
+
+        TODO: restart julia process if settings (assumed to be at
+        `pluto-deployment-environment/PlutoDeployment.toml`) change.
+        """
+        function pull(request::HTTP.Request)
+            # Need to save configuration
+            if get(ENV, "GITHUB_SECRET", "") !== ""
+                security_test = validate_github_headers(request, ENV["GITHUB_SECRET"])
+                if !security_test
+                    return HTTP.Response(501, "Not authorized!")
+                end
+            end
+
+            # github_url = get(get(JSON.parse(String(request.body)), "repository", Dict()), "html_url", nothing)
+            @async try
+                run(`git pull`)
```

Comment:
`git reset --hard` (because notebooks may change other notebooks? just be on the safe side)